examples/df.go

```diff
@@ -13,20 +14,28 @@ func formatSize(size uint64) string {
 	return sigar.FormatSize(size * 1024)
 }
 
-func df() {
+func main() {
 	fslist := sigar.FileSystemList{}
 	fslist.Get()
 
 	fmt.Fprintf(os.Stdout, output_format,
 		"Filesystem", "Size", "Used", "Avail", "Use%", "Mounted on")
 
+FSLIST:
 	for _, fs := range fslist.List {
 		dir_name := fs.DirName
-
 		usage := sigar.FileSystemUsage{}
-
 		usage.Get(dir_name)
 
+		if strings.HasPrefix(fs.SysTypeName, "fuse") {
+			continue
+		}
+		for _, hiddenPrefix := range []string{"/sys", "/proc", "/run"} {
```

Comment:
1) If we keep this code, I'd like to see a comment here mentioning why we're skipping certain volumes/filesystems. The commit comment doesn't provide a ton of justification, and a code comment would be easier to find than running `git blame` on this file to locate it.
2) Instead of ignoring `fuse` filesystems by FS type, and others by path, why not ignore all by type? Have a loop that ignores filesystems of type `fuse`, `sysfs`, `proc`,  and `tmpfs`. (And why aren't you excluding `/dev` or filesystems with type `devfs`?
3) I don't really see the value of filtering out particular filesystems or filesystem types. After changing the package name and "main" function name, running this example program on my personal Linux machine didn't produce output that was notably difficult to read and understand.

Reply:
For me this was triggered due to multiple `revokefs-fuse` lines which seem to be flatpak related.

With the filtering removed.
```
$ go run df.go 
Filesystem      Size Used Avail Use% Mounted on     
proc              0    0     0    0% /proc          
sysfs             0    0     0    0% /sys           
devtmpfs        4.0M   0   4.0M   0% /dev           
securityfs        0    0     0    0% /sys/kernel/security
tmpfs           7.6G  20M  7.6G   1% /dev/shm       
devpts            0    0     0    0% /dev/pts       
tmpfs           3.1G 2.6M  3.1G   1% /run           
cgroup2           0    0     0    0% /sys/fs/cgroup 
pstore            0    0     0    0% /sys/fs/pstore 
efivarfs          0    0     0    0% /sys/firmware/efi/efivars
bpf               0    0     0    0% /sys/fs/bpf    
/dev/mapper/luks-0a8e9c37-cf44-450e-ba85-8568ff6197f0 952G 350G  601G  37% /              
selinuxfs         0    0     0    0% /sys/fs/selinux
systemd-1         0    0     0    0% /proc/sys/fs/binfmt_misc
mqueue            0    0     0    0% /dev/mqueue    
hugetlbfs         0    0     0    0% /dev/hugepages 
debugfs           0    0     0    0% /sys/kernel/debug
tracefs           0    0     0    0% /sys/kernel/tracing
fusectl           0    0     0    0% /sys/fs/fuse/connections
configfs          0    0     0    0% /sys/kernel/config
tmpfs           7.6G 1.8G  5.9G  24% /tmp           
/dev/mapper/luks-0a8e9c37-cf44-450e-ba85-8568ff6197f0 952G 350G  601G  37% /home          
/dev/nvme0n1p2  1.0G 259M  647M 28.999999999999996% /boot          
/dev/nvme0n1p1  599M  45M  554M   8% /boot/efi      
sunrpc            0    0     0    0% /var/lib/nfs/rpc_pipefs
tmpfs           1.5G 7.5M  1.5G   1% /run/user/1000 
gvfsd-fuse        0    0     0    0% /run/user/1000/gvfs
portal            0    0     0    0% /run/user/1000/doc
binfmt_misc       0    0     0    0% /proc/sys/fs/binfmt_misc
revokefs-fuse   952G 350G  601G  37% /var/tmp/flatpak-cache-QBRCY1/org.gnome.Platform-8EWRY1
revokefs-fuse   952G 350G  601G  37% /var/tmp/flatpak-cache-QBRCY1/org.gnome.Platform-VU62Y1
revokefs-fuse   952G 350G  601G  37% /var/tmp/flatpak-cache-QBRCY1/org.gnome.Platform-V2K2Y1
revokefs-fuse   952G 350G  601G  37% /var/tmp/flatpak-cache-QBRCY1/org.gnome.Platform-SQ3WY1
revokefs-fuse   952G 350G  601G  37% /var/tmp/flatpak-cache-QBRCY1/org.gnome.Platform-LC9OY1
revokefs-fuse   952G 350G  601G  37% /var/tmp/flatpak-cache-QBRCY1/org.gnome.Platform-IY9SY1
revokefs-fuse   952G 350G  601G  37% /var/tmp/flatpak-cache-QBRCY1/org.gnome.Platform-IAZXY1
revokefs-fuse   952G 350G  601G  37% /var/tmp/flatpak-cache-QBRCY1/org.gnome.Platform-2TQOY1
revokefs-fuse   952G 350G  601G  37% /var/tmp/flatpak-cache-QBRCY1/org.gnome.Platform-HLN0Y1
tracefs           0    0     0    0% /sys/kernel/debug/tracing
```

With the filtering:
```
$ go run df.go 
Filesystem      Size Used Avail Use% Mounted on     
devtmpfs        4.0M   0   4.0M   0% /dev           
tmpfs           7.6G  20M  7.6G   1% /dev/shm       
devpts            0    0     0    0% /dev/pts       
/dev/mapper/luks-0a8e9c37-cf44-450e-ba85-8568ff6197f0 952G 350G  601G  37% /              
mqueue            0    0     0    0% /dev/mqueue    
hugetlbfs         0    0     0    0% /dev/hugepages 
tmpfs           7.6G 1.8G  5.9G  24% /tmp           
/dev/mapper/luks-0a8e9c37-cf44-450e-ba85-8568ff6197f0 952G 350G  601G  37% /home          
/dev/nvme0n1p2  1.0G 259M  647M 28.999999999999996% /boot          
/dev/nvme0n1p1  599M  45M  554M   8% /boot/efi      
sunrpc            0    0     0    0% /var/lib/nfs/rpc_pipefs
```

Adding `/dev` as prefix seems to reduce the noise even more.

I used the fs type for fuse as for me that's mounted on `/var/tmp/...` which is more likely to also have 'real' filesystems.

Reply:
On Fedora 37 `df` reports `/dev/dm-0` instead of the longer luks filenames.

This is just following a symlink:
```
$ ls -lh /dev/mapper/luks-0a8e9c37-cf44-450e-ba85-8568ff6197f0
lrwxrwxrwx. 1 root root 7 Jan 10 16:54 /dev/mapper/luks-0a8e9c37-cf44-450e-ba85-8568ff6197f0 -> ../dm-0
```

To do this in `df.go`:
```patch
diff --git a/examples/df.go b/examples/df.go
index e2d5ab2..90c7343 100644
--- a/examples/df.go
+++ b/examples/df.go
@@ -3,6 +3,7 @@ package main
 import (
        "fmt"
        "os"
+       "path/filepath"
        "strings"
 
        sigar "github.com/cloudfoundry/gosigar"
@@ -36,8 +37,10 @@ FSLIST:
                        }
                }
 
+               devResolved, _ := filepath.EvalSymlinks(fs.DevName)
+
                fmt.Fprintf(os.Stdout, output_format,
-                       fs.DevName,
+                       devResolved,
                        formatSize(usage.Total),
                        formatSize(usage.Used),
                        formatSize(usage.Avail),
```

```
$ go run df.go 
Filesystem      Size Used Avail Use% Mounted on     
                4.0M   0   4.0M   0% /dev           
                7.6G  20M  7.6G   1% /dev/shm       
                  0    0     0    0% /dev/pts       
/dev/dm-0       952G 350G  601G  37% /              
                  0    0     0    0% /dev/mqueue    
                  0    0     0    0% /dev/hugepages 
                7.6G 1.8G  5.9G  24% /tmp           
/dev/dm-0       952G 350G  601G  37% /home          
/dev/nvme0n1p2  1.0G 259M  647M 28.999999999999996% /boot          
/dev/nvme0n1p1  599M  45M  554M   8% /boot/efi      
                  0    0     0    0% /var/lib/nfs/rpc_pipefs
```

Reply:
Yeah, given that information, I'd rather either
* Remove the filtering.
* Leave the filtering in as an example of how to use some members of the relevant structures, but default the filters to "filter nothing", and populate the filters via command-line arguments.

Reply:
We could have a somewhat nice output with slightly more complexity or have a really simple and basic one. Both are fine with me.

Reply:
> +               devResolved, _ := filepath.EvalSymlinks(fs.DevName)

Yeah, let's definitely _not_ do that. The longer name is more descriptive. (Roughly noone knows what /dev/dm-0 is at a glance, but `luks-0a8e9c37...` tells you that that is something that's managed by `dm-crypt` or similar.)

Reply:
I've pushed baef0e981e22578aa5b16fe96b45d3cdc8659665 with:

```
$ df -h -t ext4
Filesystem      Size  Used Avail Use% Mounted on
/dev/nvme0n1p2  974M  260M  648M  29% /boot
$ go run df.go -t ext4
Filesystem      Size Used Avail Use% Mounted on     
/dev/nvme0n1p2  1.0G 259M  647M 28.999999999999996% /boot  
```